File: src/com/colacelli/irclib/connection/listeners/OnPartListener.java
package com.colacelli.irclib.connection.listeners;

import com.colacelli.irclib.actors.Channel;
import com.colacelli.irclib.actors.User;
import com.colacelli.irclib.connection.Connection;

public interface OnPartListener extends Listener {
    Type TYPE = Type.PART;

    void onPart(Connection connection, User user, Channel channel);

    @Override
    default Type getType() {
        return TYPE;
    }
}

File: src/main/kotlin/com/colacelli/irclib/connection/listeners/OnPartListener.kt
package com.colacelli.irclib.connection.listeners

import com.colacelli.irclib.actors.Channel
import com.colacelli.irclib.actors.User
import com.colacelli.irclib.connection.Connection

interface OnPartListener : Listener {
    fun onPart(connection: Connection, user: User, channel: Channel)

    override val type: Listener.Type
        get() = TYPE

    companion object {
        val TYPE = Listener.Type.PART
    }
}
